src/etc/inc/captiveportal.inc

```diff
@@ -2289,8 +2318,9 @@ function captiveportal_update_entry($sessionid, $new_value, $field_to_update) {
 	captiveportal_write_db("UPDATE captiveportal SET {$field_to_update} = {$new_value} WHERE sessionid = '{$sessionid}'");
 }
 
-function portal_allow($clientip, $clientmac, $username, $password = null, $attributes = null, $pipeno = null, $authmethod = null, $context = 'first') {
-	global $redirurl, $g, $config, $type, $_POST, $cpzone, $cpzoneid;
+function portal_allow($clientip, $clientmac, $username, $password = null, $attributes = null, $pipeno = null, $authmethod = null,
+ $context = 'first', $existing_sessionid = null) {
```

Comment:
Looks like this was broken up into two lines unintentionally. If it was on purpose, then it needs the indentation fixed, and balance the contents of the lines so that the first line isn't so long.